server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -63,14 +70,12 @@ public class SearchBackpressureService extends AbstractLifecycleComponent
     private final LongSupplier timeNanosSupplier;
 
     private final List<NodeDuressTracker> nodeDuressTrackers;
-    private final List<TaskResourceUsageTracker> taskResourceUsageTrackers;
+    private final Map<Class<? extends SearchBackpressureTask>, List<TaskResourceUsageTracker>> taskTrackers;
 
-    private final AtomicReference<TokenBucket> taskCancellationRateLimiter = new AtomicReference<>();
-    private final AtomicReference<TokenBucket> taskCancellationRatioLimiter = new AtomicReference<>();
+    private final Map<Class<? extends SearchBackpressureTask>, AtomicReference<TokenBucket>> rateLimiters;
+    private final Map<Class<? extends SearchBackpressureTask>, AtomicReference<TokenBucket>> ratioLimiters;
```

Comment:
Now that rate/ratio limiter is not commonly used, do you think it makes more sense to move it inside the respective `SearchBackpressureState` of each task type? That will also help remove clutter of various listeners this class has to implement as a result of updating the TokenBucket on cluster settings' changes.